micro_sam/util.py

```diff
@@ -644,46 +646,47 @@ def precompute_image_embeddings(
             where the return value is the (potentially updated) embedding save path.
     """
     ndim = input_.ndim if ndim is None else ndim
-    if tile_shape is not None:
-        assert save_path is not None, "Tiled prediction is only supported when the embeddings are saved to file."
+    # if tile_shape is not None:
+    #     assert save_path is not None, "Tiled prediction is only supported when the embeddings are saved to file."
 
     if save_path is not None:
         save_path = str(save_path)
-        data_signature = _compute_data_signature(input_)
-
         f = zarr.open(save_path, "a")
-        key_vals = [
-            ("data_signature", data_signature),
-            ("tile_shape", tile_shape if tile_shape is None else list(tile_shape)),
-            ("halo", halo if halo is None else list(halo)),
-            ("model_type", predictor.model_type)
-        ]
-        if "input_size" in f.attrs:  # we have computed the embeddings already and perform checks
-            for key, val in key_vals:
-                if val is None:
-                    continue
-                # check whether the key signature does not match or is not in the file
-                if key not in f.attrs or f.attrs[key] != val:
-                    raise RuntimeError(
-                        f"Embeddings file {save_path} is invalid due to unmatching {key}: "
-                        f"{f.attrs.get(key)} != {val}.Please recompute embeddings in a new file."
-                    )
-                    if wrong_file_callback is not None:
-                        save_path = wrong_file_callback(save_path)
-                        f = zarr.open(save_path, "a")
-                    break
-
+    else:
+        f = zarr.group()
+    data_signature = _compute_data_signature(input_)        
```

Comment:
We should only do this if we actually store the embeddings to a zarr file. Otherwise we do some unnecessary computations.